src/twisted/test/test_iosim.py

```diff
@@ -39,3 +44,85 @@ def test_writeSequence(self):
         a.writeSequence([b"b", b"c", b"d"])
 
         self.assertEqual(b"".join(a.stream), b"abcd")
+
+
+
+@implementer(IPushProducer)
+class StrictPushProducer(object):
```

Comment:
Yea, without the ability to customize this exception at all, I don't think it's appropriate.  This is a good exception to tell the state machine maintainer that they screwed up.  Otherwise, it's an implementation detail leaking out.